resources/providers/gcplib/auth/credentials.go

```diff
@@ -49,37 +53,47 @@ type ConfigProvider struct {
 }
 
 func (p *ConfigProvider) GetGcpClientConfig(ctx context.Context, cfg config.GcpConfig, log *logp.Logger) (*GcpFactoryConfig, error) {
+	// TODO: is this for development only? if so add a comment
```

Comment:
Not for development, it's for cases where we fetch the ADC from the service account attached to compute instance, the one running elastic agent.